### DISCUSSION TITLE
GpsFactor that includes orientation

### DISCUSSION THREAD

**Discussion by leo-stan** [borglab/gtsam#1299] [Q&A] [answered]
Hey all,

I want to add a factor to some values with their position and orientation, basically just like a GpsFactor but that includes orientation (so I guess using a full Pose3 rather than just a Point3?). The idea is that every now and then when my robot is moving I could provide a ground-truth of the full Pose3 and optimize the poses in between.

Any help on where to start would be fantastic!

Cheers,
Leo

**Comment by varunagrawal** [comment, marked as answer]
Perhaps the `PriorFactor<Pose3>` is what you're looking for?
